### PR TITLE
fix: increase cache for Inter font file [sc-12301]

### PR DESCRIPTION
**Pull request by ndom91**
## Affected Components
* [ ] Content & Marketing
* [ ] Test
* [ ] Docs
* [ ] Learn
* [x] Other

## Pre-Requisites
* [x] Code is linted (`npm run lint`)

## Notes for the Reviewer

- ~Moves `Inter.var.woff2` off of the self-hosting setup to requiring it from Google Fonts. This is due to bandwidth concerns, as that Inter font file was our `#1` bandwidth consumer from Vercel (which charges an extreme amount for bandwidth).~

- Changed to increase cache headers for the self-hosted Inter font file.

Last 14 Days:
image

See discussion with Daniel in Slack: https://checklyhq.slack.com/archives/C02CES5S4LS/p1666860610470859

### CODE REVIEW

**Review by nahuelon — DISMISSED** [review]
LGTM.

**Comment by hlenke** [comment]
What's the lighthouse impact of this change? Did you test that? - Hannes

------------------------------------
*Hannes Lenke*
*Lindenpromenade 6*
*15344 Strausberg*
*
*
*EMail: ***@***.****
*Phone: (+49) 175 5460664*

**Review by ianaya89 — APPROVED** [review]

File: src/scss/checkly-webapp-src/style/fonts/fonts.scss [hunk 1]
```diff
@@ -1,10 +1,4 @@
-@font-face {
-	font-family: 'Inter';
-	font-style: normal;
-	font-weight: 100 900;
-	font-display: optional;
-	src: url('../fonts/Inter/Inter.var.woff2') format('woff2');
-}
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');
```
Comment (stefanjudis) [inline src/scss/checkly-webapp-src/style/fonts/fonts.scss:1]: @ndom91 Using `import` is a pretty bad perf pattern. I'm not sure where this pops out, but are we sure we want to do this? It potentially delays first paint.

**Comment by stefanjudis** [comment]
@ndom91 Also these caching settings for the font don't look alright to me. If we're not caching the font on the client it's not surprising, that it's driving so much bandwidth. 

Generally speaking, self-hosting fonts is the way to go these days. 🙈 

image

File: src/scss/checkly-webapp-src/style/fonts/fonts.scss (hunk 1 above)
Comment (ndom91) [reply src/scss/checkly-webapp-src/style/fonts/fonts.scss:1]: Yeah so this is what I mentioned in Slack, not-selfhosting your fonts forces us to do another import, make another https connection (incl tls handshake), etc. 

It kinda goes against "best-practices", but also would save us on bandwidth costs. (This inter file is our biggest bandwidth perpetrator). 

So its up to you guys, I dont really have a strong opinion. Was just trying to improve bandwidth costs with these marketing website changes  :+1: (See discussion with Daniel linked above for more details).

CC: @hlenke

**Comment by ndom91** [comment]
> @ndom91 Also these caching settings for the font don't look alright to me. If we're not caching the font on the client it's not surprising, that it's driving so much bandwidth.
> 
> Generally speaking, self-hosting fonts is the way to go these days. see_no_evil
> 
> image

Yeah so it seems to be being built as if it were a code asset, i.e. like a css file, and appended a suffix for cache-busting, etc.

I'm not familiar enough with Hugo to say off the top of my head where this is coming from exactly, but I'm sure we can change it to tell it this is a static asset that never changes and cache client-side for as long as possible.

That seems to be a good middle-ground here. 

Good eye!

**Comment by ndom91** [comment]
@stefanjudis the Inter font file now gets loaded without a hash appended to the filename and with proper cache headers :partying_face: 

image

**Review by stefanjudis — APPROVED** [review]
Great stuff. LGTM.

Out of curiosity, why did you decide to drop the file hashes? If I understood correctly, it was just a matter of `vercel.json`?

Also, let's set a reminder to check in in 2w to see if this helped?

**Comment by ndom91** [comment]
I dropped the hashes because the hash would be different every deploy, so even if we set the cache header to cache it client-side for 1 month, for example, then as soon as we make a new deploy the filename would be different and it'll need to be freshly downloaded again. Right?

**Comment by stefanjudis** [comment]
> then as soon as we make a new deploy the filename would be different and it'll need to be freshly downloaded again

🤔  Aren't webpack file hashes file content dependent? So that the same file always has the same hash?

Anyways. Case closed. :D Thanks for handling this Nico!

**Pull request merged**